app/src/renderer/components/common/ActionModal.vue

```diff
@@ -0,0 +1,280 @@
+        <slot></slot>
+
+        <hr />
+
+        <tm-form-group
+          class="action-modal-form-group"
+          field-id="sign-method"
+          field-label="Signing Method"
+        >
+          <tm-field
+            id="sign-method"
+            v-model="selectedSignMethod"
+            :options="signMethods"
+            type="select"
+          />
+        </tm-form-group>
+
+        <tm-form-group
+          v-if="selectedSignMethod === `local`"
+          :error="$v.password.$error && $v.password.$invalid"
+          class="action-modal-group"
+          field-id="password"
+          field-label="Password"
+        >
+          <tm-field
+            id="password"
+            v-model="password"
+            type="password"
+            placeholder="Password"
+          />
+          <tm-form-msg
+            v-if="$v.password.$error && !$v.password.required"
+            name="Password"
+            type="required"
+          />
+        </tm-form-group>
+      </div>
+      <div class="action-modal-footer">
+        <slot name="action-modal-footer">
+          <tm-form-group class="action-modal-group">
+            <div class="action-modal-footer">
+              <tm-btn
+                v-if="sending"
+                value="Sending..."
+                disabled="disabled"
+                color="primary"
+              />
+              <tm-btn
+                v-else-if="!connected"
+                value="Connecting..."
+                disabled="disabled"
+                color="primary"
+              />
+              <tm-btn
+                v-else
+                id="cast-vote"
+                color="primary"
+                value="Submit"
+                @click.native="validateForm"
+              />
+            </div>
+          </tm-form-group>
+        </slot>
+        <p
+          v-if="submissionError"
+          class="tm-form-msg sm tm-form-msg--error submission-error"
+        >
+          {{ submissionError }}
+        </p>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script>
+import ClickOutside from "vue-click-outside"
+import TmBtn from "common/TmBtn"
+import TmField from "common/TmField"
+import TmFormGroup from "common/TmFormGroup"
+import TmFormMsg from "common/TmFormMsg"
+import { mapGetters } from "vuex"
+import { requiredIf } from "vuelidate/lib/validators"
+
+export default {
+  name: `action-modal`,
+  directives: {
+    ClickOutside
+  },
+  components: {
+    TmBtn,
+    TmField,
+    TmFormGroup,
+    TmFormMsg
+  },
+  props: {
+    title: {
+      type: String,
+      required: true
+    },
+    submitFn: {
+      type: Function,
+      required: true
+    },
+    validate: {
+      type: Function,
+      required: true
+    },
+    submissionErrorPrefix: {
+      type: String,
+      default: `Submitting data failed`
+    }
+  },
+  data: () => ({
+    signMethod: null,
+    password: null,
+    selectedSignMethod: `local`,
+    signMethods: [
```

Comment:
hmmm, we will still need this in the development setup. and it might be nice to indicate that there will be more signmethods coming. but I see the point. what's your take?

Reply:
yeah it's ok forget i said it.